noty-android/app/src/main/java/dev/shreyaspatil/noty/utils/NetworkUtils.kt

```diff
@@ -67,21 +67,14 @@ val Context.currentConnectivityState: ConnectionState
     }
 
 private fun getCurrentConnectivityState(connectivityManager: ConnectivityManager): ConnectionState {
-    var currentState: ConnectionState = ConnectionState.Unavailable
-
     // Retrieve current status of connectivity
-    connectivityManager.allNetworks.forEach { network ->
-        val networkCapability = connectivityManager.getNetworkCapabilities(network)
-
-        networkCapability?.let {
-            if (it.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET)) {
-                currentState = ConnectionState.Available
-                return@forEach
```

Comment:
This doesn't break out of the `forEach` iteration. Instead, it's a local return inside the lambda that's passed to `forEach`. The logic inside `forEach` will continue iterating through the collection and invoking this lambda over and over again even after this statement has been executed (which I'm pretty sure is not what's expected to happen here).

Try [this simple example](https://pl.kotl.in/OXCm3Be0f) that shows this issue.

Reply:
Thanks for detailed explanation. Appreciapted :+1